sshd-core/src/test/java/org/apache/sshd/common/config/keys/OpenSSHKeyUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.md5,
+                "MD5:fb:29:14:8d:94:f9:1d:cf:6b:0e:a4:35:1d:83:44:2f"
+            ),
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.sha256,
+                "SHA256:grxw4KhY1cK6eOczBWs7tDVvo9V0PQw4E1wN1gJvHlw"
+            )
+        ),
+        new KeyEntry(
+            "ecdsa-sha2-nistp384 AAAAE2VjZHNhLXNoYTItbmlzdHAzODQAAAAIbmlzdHAzODQAAABhBFImZtcTj842stlcVHLFBFxTEx7lu3jW9aZCvd0r9fUNKZ6LbRPh6l1oJ4ozArnw7XreQBUc5oNd9HB5RNJ8jl1nWXY5cXBA7McZrKZrYmk+zxNhH6UL+kMLaJkyngJHQw== root@osv-linux",
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.md5,
+                "MD5:e6:dc:a2:4f:5b:11:b2:3c:0f:e8:f6:d8:d1:01:e9:d3"
+            ),
+            new Pair<>(
+                (DigestFactory)BuiltinDigests.sha512,
+                "SHA512:4w6ZB78tmFWhpN2J50Ok6WeMJhZp1X0xN0EKWxZmRLcYDbCWhyJDe8lgrQKWqdTCMZ5aNEBl9xQUklcC5Gt2jg"
+            )
+        )
+    };
+
+    public OpenSSHKeyUtilsTest() {
+        super();
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws IOException, GeneralSecurityException {
+        for (KeyEntry entry : KEY_ENTRIES) {
+            try {
+                entry.key = PublicKeyEntry.parsePublicKeyEntry(entry.keyString).resolvePublicKey();
+            } catch (NoSuchProviderException e) {
+                // ignore key
+            }
+        }
+    }
+
+    @Test
+    public void testFingerprint() throws Exception {
+        for (KeyEntry kentry : KEY_ENTRIES) {
+            if (kentry.key != null) {
+                for (Pair<DigestFactory, String> dentry : kentry.hashes) {
+                    assertEquals(
+                        String.format("Fingerprint does not match for key %s digest %s", kentry.key.getAlgorithm(), dentry.getFirst().getName()),
+                        dentry.getSecond(),
+                        OpenSSHKeyUtils.getFingerPrint(dentry.getFirst(), kentry.key)
+                    );
+                    assertEquals(
+                        String.format("Fingerprint check failed for key %s digest %s", kentry.key.getAlgorithm(), dentry.getFirst().getName()),
+                        new Pair<Boolean, String>(true, dentry.getSecond()),
+                        OpenSSHKeyUtils.checkFingerPrint(dentry.getSecond(), kentry.key)
+                    );
+                    assertEquals(
+                        String.format("Fingerprint check succeeded for key %s invalid digest %s", kentry.key.getAlgorithm(), dentry.getFirst().getName()),
+                        new Pair<Boolean, String>(false, dentry.getSecond()),
+                        OpenSSHKeyUtils.checkFingerPrint(dentry.getSecond() + "A", kentry.key)
+                    );
+                }
+            }
+        }
+    }
```

Comment:
I am not a big fan of repeated "getXXX" calls to the same value. Preferably:
```java
PublicKey key = kentry.key;
if (key != null) {
    String algo = key.getAlgorithm();
    for (Pair<DigestFactory, String> dentry : kentry.hashes) {
        DigestFactory factory = dentry.getFirst();
        String name = factory.getName();
        String expected = dentry.getSecond();
        assertEquals(
                  String.format("Fingerprint does not match for key %s digest %s", algo, name),
                  expected, OpenSSHKeyUtils.getFingerPrint(factory, key));
         ...the other asserts...
    }
}
```
Easier to read and understand instead of having to remember what "getFirst" and "getSecond" means